base/client.jl

```diff
@@ -221,45 +221,45 @@ function process_options(opts::JLOptions)
 
         # startup worker
         if opts.worker != C_NULL
-            start_worker(String(opts.worker)) # does not return
+            start_worker(unsafe_string(opts.worker)) # does not return
         end
         # add processors
         if opts.nprocs > 0
             addprocs(opts.nprocs)
         end
         # load processes from machine file
         if opts.machinefile != C_NULL
-            addprocs(load_machine_file(String(opts.machinefile)))
+            addprocs(load_machine_file(unsafe_string(opts.machinefile)))
         end
         # load file immediately on all processors
         if opts.load != C_NULL
             @sync for p in procs()
-                @async remotecall_fetch(include, p, String(opts.load))
+                @async remotecall_fetch(include, p, unsafe_string(opts.load))
             end
         end
         # eval expression
         if opts.eval != C_NULL
             repl = false
-            eval(Main, parse_input_line(String(opts.eval)))
+            eval(Main, parse_input_line(unsafe_string(opts.eval)))
             break
         end
         # eval expression and show result
         if opts.print != C_NULL
             repl = false
-            show(eval(Main, parse_input_line(String(opts.print))))
+            show(eval(Main, parse_input_line(unsafe_string(opts.print))))
             println()
             break
         end
         # eval expression but don't disable interactive mode
         if opts.postboot != C_NULL
-            eval(Main, parse_input_line(String(opts.postboot)))
+            eval(Main, parse_input_line(unsafe_string(opts.postboot)))
         end
         # load file
         if !isempty(ARGS) && !isempty(ARGS[1])
             # program
             repl = false
             # remove filename from ARGS
-            global PROGRAM_FILE = String(shift!(ARGS))
+            global PROGRAM_FILE = shift!(ARGS)
```

Comment:
`ARGS` is a `Vector{String}`, a typeassert seems superfluous.  The only reason for the `String` call here was that this used to be `UTF8String(shift!(ARGS))` and in those days the conversion was needed to avoid having the type of `PROGRAM_FILE` depend on the arguments.   I think it just got auto search-and-replaced by `String` when @StefanKarpinski made the switch.